### PR TITLE
fix(no-tx-addr): added custom error

### PR DESCRIPTION
**Pull request by BLuEScioN**
Issue https://github.com/hirosystems/explorer/issues/1550

image

We already had a message for when there are no txs. It just wasn't showing. There is a lot of deep logic around displaying the error/no txs message here, from capturing the error at the error boundary to just handling no txs and how that is displayed. I think it would be best to use what we already have for the "no txs message" and don't make any changes until after the redesign.

ST2Y2SFNVZBT8SSZ00XXKH930MCN0RFREB2GQG7CJ

### CODE REVIEW

**Comment by BLuEScioN** [comment]
> It works great when navigating directly to the address but not searching. Can we do something about it?

You can still search for it and the search functionality behaves exactly the same as for a address that actually exists. What exactly did you think needs to be changed?

image

image

**Review by He1DAr — APPROVED** [review]

File: src/app/address/[principal]/PageClient.tsx [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
 import { StxBalance } from './StxBalance';
 import { TokenBalanceCard } from './TokenBalanceCard';
-import { Wrapper } from './Wrapper';
+
+export function AddressPageLayout(props: GridProps) {
+  return (
+    <Grid
+      gridColumnGap="32px"
+      gridTemplateColumns={['100%', '100%', 'repeat(1, calc(100% - 352px) 320px)']}
+      gridRowGap={['32px', '32px', 'unset']}
```
Comment (He1DAr) [inline src/app/address/[principal]/PageClient.tsx:19]: would you mind using Chakra variables for the gap?

File: src/common/hooks/useInfiniteQueryResult.ts [hunk 2]
```diff
@@ -31,11 +31,14 @@ export function useSuspenseInfiniteQueryResult<T>(
...
-    );
+    const pages = response.data?.pages;
+    const results = pages?.map(page => page.results);
+    const flatResults = results?.flat();
+    const slicedResults = flatResults?.slice(0, limit);
+    return slicedResults || [];
+    // return response.data?.pages
+    //   .map(page => page.results)
+    //   .flat()
+    //   .slice(0, limit) || []
```
Comment (He1DAr) [inline src/common/hooks/useInfiniteQueryResult.ts:42]:
```suggestion

```

File: src/app/address/[principal]/PageClient.tsx (hunk 1 above)
Comment (BLuEScioN) [reply src/app/address/[principal]/PageClient.tsx:19]: done

File: src/app/address/[principal]/TokenBalanceCard/FtBalance.tsx [hunk 3]
```diff
@@ -20,13 +20,13 @@ const TokenAssetListItem = dynamic(
...
-  const FTokens = Object.keys(balance.fungible_tokens).filter(
+  const ft = Object.keys(balance.fungible_tokens).filter(
     key => Number(balance.fungible_tokens[key]?.balance) > 0
   );
 
-  return FTokens.length ? (
+  return ft.length > 0 ? (
     <>
-      {FTokens?.map((key, index, arr) => (
+      {ft?.map((key, index, arr) => (
```
[the remark this answers is not in the document]
Comment (BLuEScioN) [reply src/app/address/[principal]/TokenBalanceCard/FtBalance.tsx:29]: done

File: src/common/hooks/useInfiniteQueryResult.ts (hunk 2 above)
Comment (BLuEScioN) [reply src/common/hooks/useInfiniteQueryResult.ts:42]: done

**Pull request merged**